kotlin/workflow-runtime/src/main/java/com/squareup/workflow/debugging/WorkflowUpdateDebugInfo.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.debugging
+
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Kind
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Kind.ChildDidUpdate
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Kind.DidUpdate
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Source
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Source.External
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Source.Subtree
+import com.squareup.workflow.debugging.WorkflowUpdateDebugInfo.Source.Worker
+import com.squareup.workflow.internal.WorkflowId
+
+/**
+ * A description of a workflow update triggered by a [Source] (worker, event, etc).
+ *
+ * This is a simple linked list that represents a traversal down the workflow tree that starts at
+ * the root and indicates, for each workflow, if its child output something that it handled
+ * (see [Kind]), or if it was just a parent of a workflow that didn't output anything.
+ *
+ * When a workflow handles an update, the type of update is indicated by [Source].
+ */
+data class WorkflowUpdateDebugInfo(
+  val workflowType: String,
+  val kind: Kind
+) {
+  constructor(
+    workflowId: WorkflowId<*, *, *>,
+    kind: Kind
+  ) : this(workflowId.typeDebugString, kind)
+
+  /**
+   * A sealed class that indicates whether a workflow actually executed a `WorkflowAction`, or
+   * was just the ancestor of a workflow that did.
+   *
+   * Contains two subclasses, see their documentation for details:
+   * - [DidUpdate]
+   * - [ChildDidUpdate]
+   */
+  sealed class Kind {
+    // TODO more kotliny names
```

Comment:
"DidUpdate" is a very idiomatic name in Swift/ObjC, not so much in Kotlin. I'm thinking just `Updated` and `ChildUpdated`?

Reply:
SG